modules/nixos.nix

```diff
@@ -165,5 +164,42 @@ in {
           concatStringsSep " " ruleString
       ) (filter (f: f.enable && f.source != null) (attrValues files));
     }) (filterAttrs (_: u: u.files != {}) config.homes);
+    systemd.services = mapAttrs' (name: {files, ...}: {
+      name = "hjem-monitor-" + name;
+      value = {
+        enable = true;
+        wants = ["systemd-tmpfiles-setup.service" "nix-daemon.socket"];
+        after = ["nix-daemon.socket"];
+        wantedBy = ["default.target"];
+        description = "Monitoring for Hjem files";
+        scriptArgs = "${toString (map (
+          file: file.target
+        ) (filter (f: f.enable && f.source != null) (attrValues files)))}";
+        script = ''
+          #! ${pkgs.runtimeShell} -e
+          code=0
+          err=""
+          normal=""
+          if test -t 1; then
+            ncolors=$(tput colors)
+            if test -n "$ncolors" && test $ncolors -ge 8; then
+              err="$(tput bold)$(tput setaf 3)"
```

Comment:
This is unused, shellcheck will warn you.

Reply:
the test for colors is not needed?

Reply:
Didn't anticipate Github's comment not highlighting the specific line.

You're defining `err=...` here

```sh
if test -t 1; then
  ncolors=$(tput colors)
  if test -n "$ncolors" && test "$ncolors" -ge 8; then
    err="$(tput bold)$(tput setaf 3)"
    normal="$(tput sgr0)"
  fi
fi
```

but it's not referenced elsewhre. Is this some abstraction that I'm not aware of?

Reply:
err is used for colouring and bolding the text, and it’s first set as an empty string but later set to the tput stuff if the terminal allows for colours.

Reply:
```bash
            if [ ! -L "$var" ] ; then
              echo "''${err}$var is not managed by Hjem due to a file conflict, please move or remove the current file.''${normal}"
              code=1
            fi
```